bootloader/grub.go

```diff
@@ -55,16 +55,16 @@ type grub struct {
 // newGrub create a new Grub bootloader object
 func newGrub(rootdir string, opts *Options) Bootloader {
 	g := &grub{rootdir: rootdir}
-	if opts != nil && (opts.Recovery || opts.NoSlashBoot) {
+	if opts != nil {
+		g.uefiRunKernelExtraction = opts.Role == RoleRunMode
+		g.recovery = opts.Role == RoleRecovery
+		g.native = opts.NoSlashBoot || g.recovery
```

Comment:
mmm reading this code makes me wonder if we could also do `s/NoSlashBoot/NativeLayout/` too ?

Reply:
I don't find NativeLayout that much clearer, maybe is native that should be renamed here?

Reply:
I did that, I clarified and expanded "native" where is used but kept this flag name. The other option would be to call this Native[Host]PartitionLayout that is a bit long though.

Reply:
mmm fair enough I guess at the time I looked at this native layout stuck out as something more obvious to me than "no slash boot", but maybe that's because I don't usually associated "/" with the word slash, but anyways probably fine to ignore this

Reply:
whoops seems I commented literally 12 seconds after you did... what you ended up doing is fine